portal-api/src/v1/routes.js

```diff
@@ -75,62 +84,62 @@ authRouter.route('/mandatory-criteria/latest').get(mandatoryCriteria.findLatest)
 authRouter
   .route('/mandatory-criteria/:version')
   .get(mandatoryCriteria.findOne)
-  .delete(validate({ role: ['editor'] }), mandatoryCriteria.delete);
+  .delete(validateUserHasSufficientRole({ allowedNonAdminRoles: [EDITOR] }), mandatoryCriteria.delete);
 
 authRouter.route('/users').get(users.list).post(users.create);
 authRouter.route('/users/:_id').get(users.findById).put(users.updateById).delete(users.remove);
 authRouter.route('/users/:_id/disable').delete(users.disable);
 
 authRouter.use('/gef', gef);
 
-authRouter.route('/deals').post(validate({ role: ['maker'] }), dealsController.create);
-authRouter.route('/deals').get(validate({ role: ['maker', 'checker', 'admin'] }), dealsController.getQueryAllDeals);
+authRouter.route('/deals').post(validateUserHasSufficientRole({ allowedNonAdminRoles: [MAKER] }), dealsController.create);
+authRouter.route('/deals').get(validateUserHasSufficientRole({ allowedNonAdminRoles: [MAKER, CHECKER, READ_ONLY, ADMIN] }), dealsController.getQueryAllDeals);
```

Comment:
If we need `ADMIN` on these endpoints then should we rename the property to `acceptableRoles` as `ADMIN` and `allowedNonAdminRoles` contract.

Same applies to all of the remaining occurrences on this file.

Reply:
Good point on the naming but I don't think it's worth changing because it will make more sense after #1975 (which changes the "I can access any route" role to `admin` so we don't need to specify it anymore